eth/backend.go

```diff
@@ -260,8 +260,20 @@ func New(stack *node.Node, config *ethconfig.Config) (*Ethereum, error) {
 	// Start the RPC service
 	eth.netRPCService = ethapi.NewNetAPI(eth.p2pServer, config.NetworkId)
 
+	apis := eth.APIs()
```

Comment:
There is a corresponding method in  `les/client.go`. Same filter is needed there too. 

However, this strikes me as a bit unorthodox way to filter. By default, `personal` is disabled. It only becomes enabled if someone lists it as enabled rpc, _or_ it's over `ipc`, so everything is exposed. 

Why not blacklist the module, so that it becomes enabled IFF: 
- it's explicitly enabled by e.g `http.api=personal`, AND `rpc.enablepersonal`, OR
- we're on `ipc` AND `rpc.enablepersonal`,

Then the filtering would happen in `node.go`, in `startRPC()`. Doing it there seems more in line with how it's been done before... ?

Reply:
The problem is that the personal namespace is constructed (and needs to be because of the address locker) on `ethapi.GetAPIs()`. So it is enabled on ipc and js console always, only disabled on http (if `http.api` doesn't include personal). How do you mean blacklisting the module?

Reply:
Well, the `ipc` is started a bit before startHttp
```
func (n *Node) startRPC() error {
	if err := n.startInProc(); err != nil {
		return err
	}
```
And it just assumes that all are enabled: 
```
func (n *Node) startInProc() error {
	for _, api := range n.rpcAPIs {
		if err := n.inprocHandler.RegisterName(api.Namespace, api.Service); err != nil {

```
But if change that handling a bit, we could filter the `ipc` similarly as we already do with `http`/`ws`. As for `js`, I'm not sure.